addons/sale_pdf_quote_builder/utils.py

```diff
@@ -2,11 +2,27 @@
 
 import base64
 import io
+import re
 
 from odoo import _
 from odoo.exceptions import ValidationError
 from odoo.tools import pdf
 
+_sol_field_pat = re.compile(r'^sol_id_(\d+)__(.+)$')
+
+
+def is_sol(field):
+    return bool(re.match(_sol_field_pat, field))
+
+
+def sol_prefix_fn(sol_id):
+    return lambda field: f'sol_id_{sol_id}__{field}'
```

Comment:
Seems a bit overkill to put this as a method in the utils, when it's used only once, no need to extract that, it'd be clearer in the code to know exactly what's happening without having to look in the utils

Reply:
fair, tho it is used twice, its use in the test is why I moved it to utils

Reply:
Oh right, the reason I added it to utils is because it's where I had the `sol_field` pattern defined. That way, if the format ever needs changing for some reason, it could be done in 2 lines in 1 file (+ the test).

As long as there's clear ways to check, add & remove the marker, its exact format should be irrelevant to the rest of the code imo.

I could move them back to the `models/ir_actions_report.py` file, it just looked kinda odd to have such minor utility functions defined above the `IrActionsReport` class.